### ISSUE TITLE
`eq-without-hash` (`PLW1641`) does not check for `__hash__` in superclasses

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#6932]
i noticed this is mentioned [in the rule's docs](https://beta.ruff.rs/docs/rules/eq-without-hash/) but i'm raising it because i couldn't see an issue for it

**Comment by charliermarsh** [comment]
👍 Yeah this is why the rule is in the nursery -- as long as we don't support multi-file analysis, we can't implement it reliably. I'm gonna close as this is basically a known issue that will eventually be solved by multi-file analysis.

**Comment by DetachHead** [comment]
wouldn't it make sense to keep the issue open to track it if it's a known issue?

**Comment by LefterisJP** [comment]
> We could consider creating a central tracking issue (or a tag like @zanieb did here). I just want to avoid piling up a bunch of known issues related to multi-file analysis that can't be worked on individually and make it harder to navigate the issue tracker.

Also hit this one while enabling `--preview`. It's a useful rule. It found out 1 place we were missing `__hash__` but then had 6 false positives due to superclasses.

Added ```# noqa: PLW1641  # hash implemented by superclass``` at the false positives for now and keeping the rule.

**Comment by DetachHead** [comment]
> We could consider creating a central tracking issue

raised #7447

**Comment by KotlinIsland** [comment]
- see #14883, it shouldn't look in superclasses
